sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/BooleanSimplificationSuite.scala

```diff
@@ -160,4 +166,12 @@ class BooleanSimplificationSuite extends PlanTest with PredicateHelper {
       testRelation.where('a > 2 || ('b > 3 && 'b < 5)))
     comparePlans(actual, expected)
   }
+
+  test("Complementation Laws") {
+    checkCondition('a && !'a, LocalRelation(testRelation.output, Seq.empty))
```

Comment:
Yes, but I wanted to make sure it was clear that the resulting plan node was a LocalRelation node that produced no data.